File: Java/ClimbingStairs.java
public class ClimbingStairs {
    public int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return 2;
        }

        int[] ways = new int[n];
        ways[0] = 1;
        ways[1] = 2;

        for (int i = 2; i < n; i++) {
            ways[i] = ways[i - 1] + ways[i - 2];
        }
        return ways[n - 1];
    }

    public static void main(String[] args) {
        int n = 2;
        ClimbingStairs obj = new ClimbingStairs();
        System.out.println("Distinct Ways : " + obj.climbStairs(n));
    }
}

File: Python/ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1:
            return 1
        else:
            no_of_ways=[1,2] # for n=1 and n=2
            for i in range(2,n):
                no_of_ways.append(no_of_ways[i-1]+no_of_ways[i-2])
        return no_of_ways[-1]
    

s=Solution()
n=2
print("Distinct ways : ",s.climbStairs(n))

# Input: n = 2
# Output: 2

# Input: n = 3
# Output: 3